content/log/parsing.md

```diff
@@ -0,0 +1,212 @@
+|`geoip` |parses an IP or a host and returns a JSON object that contains the continent, country, city and location of the IP address.|
+|`useragent([decodeuricomponent:true/false])`| parses a user-agent and returns a JSON object that contains the device, os and the browser represented by the agent. [More info](#useragent-parser)|
+|`querystring`| extracts all the key-value pairs in a matching URL query string (eg. "productId=superproduct&promotionCode=superpromo")|
+|`decodeuricomponent`| this core filter decodes uri components.|
+|`lowercase`| returns the lower cased string.|
+|`uppercase` |returns the upper cased string.|
+|`keyvalue([separatorStr[, characterWhiteList [, quotingStr]])` |extracts key value pattern and returns a JSON object. [More info](#key-value) |
+|`scale(factor)` | multiplies the expected numerical value by the provided factor.|
+|`array([[openCloseStr, ] separator][, subRuleOrFilter)` | parses a string sequence of tokens and returns it as an array.|
+|`url`|parses an url and returns all the tokenized members (domain, query params, port, etc) in a JSON object. [More info](#url-processor)|
+
+## Examples
+Let’s see some examples to understand better how this works.
+
+### Key value
+
+This is the key value core filter : `keyvalue([separatorStr[, characterWhiteList [, quotingStr]])`` where:
+
+* `separatorStr` : defines the separator. Default `=`
+* `characterWhiteList`: defines additional non escaped value chars. Default `\\w.\\-_@`
+* `quotingStr` : defines quotes. Default behavior which detects quotes (`<>`, `"\"\""`, ...). When defined default behavior is replaced by allowing only defined quoting char. For e.g. `<>` matches *test=<toto sda> test2=test* .
+
+Use filters such as **keyvalue()** to more-easily map strings to attribute: 
+
+log: 
+
+```
+user=john connect_date=11/08/2017 id=123 action=click
+```
+
+Rule
+
+```
+rule keyvalue("="," "))
+```
+
+{{< img src="log/parsing/parsing_example_2.png" alt="Parsing example 2" >}}
+
+You don't need to specify the name of your parameters as they were already contained in the log.
+If you add an **extract** parameter in your rule pattern you would have:
+
+{{< img src="log/parsing/parsing_example_2_bis.png" alt="Parsing example 2 bis" >}}
+
+### Parsing dates
+
+The date matcher transforms your timestamp in the EPOCH format.
+
+
+||||
+|:---|:----|:----|
+|**Raw string** | **Parsing rule** | **Result** |
+|14:20:15| `%{date("HH:mm:ss"):date}` |{"date": 51615000} |
+|11/10/2014| `%{date("dd/mm/yyyy"):date}`| { "date": 1412978400000}|
+|Thu Jun 16 08:29:03 2016 | `%{date("EEE MMM dd HH:mm:ss yyyy"):date}` | {"date": 1466065743000}|
+|06/Mar/2013:01:36:30 +0900| `%{date("dd/MMM/yyyy:hh:mm:ss Z"):date}` | {"date": 1362501390000}|
+|2016-11-29T16:21:36.431+0000| `%{date("yyyy-MM-dd'T'HH:mm:ss.SSSZ"):date}` | {"date": 1480436496431} |
+|06/Feb/2009:12:14:14.655 | `%{date("dd/MMM/yyyy:HH:mm:ss.SSS"):date}` | {“date”: 1233922454655}|
+
```

Comment:
Need to add an example with timezone.

Thu Jun 16 08:29:03 2016
rule %{date("EEE MMM dd HH:mm:ss yyyy","Europe/Paris"):date}

{
  "date": 1466058543000
}

Reply:
👍